code/modules/mob/living/silicon/pai/software.dm

```diff
@@ -114,176 +126,191 @@
+				button.color = null
 
 /mob/living/silicon/pai/Topic(href, href_list)
-	..()
+	if(..())
+		return
 	var/soft = href_list["software"]
 	var/sub = href_list["sub"]
 	if(soft)
 		screen = soft
-	if(sub)
-		subscreen = text2num(sub)
-	switch(soft)
-		// Purchasing new software
-		if("buy")
-			if(subscreen == 1)
-				var/target = href_list["buy"]
-				if(available_software.Find(target) && !software.Find(target))
-					var/cost = available_software[target]
-					if(ram >= cost)
-						software.Add(target)
-						ram -= cost
+		if(sub)
+			subscreen = text2num(sub)
+		switch(soft)
+			if("buy") // Purchasing new software
+				if(subscreen == 1)
+					var/target = href_list["buy"]
+					if(available_software.Find(target) && !software.Find(target))
+						var/cost = available_software[target]
+						if(ram >= cost)
+							software.Add(target)
+							ram -= cost
+							LoadedSoftware(target)
+						else
+							temp = "Insufficient RAM available."
+					else
+						temp = "Trunk <TT> \"[target]\"</TT> not found."
+
+
+			if("radio") // Configuring onboard radio
+				radio.attack_self(src)
+
+			if("image") // Set pAI card display face
+				var/newImage = input("Select your new display image.", "Display Image", "Happy") in list("Happy", "Cat", "Extremely Happy", "Face", "Laugh", "Off", "Sad", "Angry", "What")
+				var/pID = 1
+
+				switch(newImage)
+					if("Happy")
+						pID = 1
+					if("Cat")
+						pID = 2
+					if("Extremely Happy")
+						pID = 3
+					if("Face")
+						pID = 4
+					if("Laugh")
+						pID = 5
+					if("Off")
+						pID = 6
+					if("Sad")
+						pID = 7
+					if("Angry")
+						pID = 8
+					if("What")
+						pID = 9
+					if("Null")
+						pID = 10
+				card.setEmotion(pID)
+
+			if("news")
+				newscaster.ui_interact(src)
+
+			if("camzoom")
+				aicamera.adjust_zoom(usr)
+
+			if("signaller")
+				if(href_list["send"])
+					signaler.send_activation()
+					audible_message("[icon2html(src, hearers(src))] *beep* *beep* *beep*")
+					playsound(src, 'sound/machines/triple_beep.ogg', ASSEMBLY_BEEP_VOLUME, TRUE)
+
+				if(href_list["freq"])
+					var/new_frequency = (signaler.frequency + text2num(href_list["freq"]))
+					if(new_frequency < MIN_FREE_FREQ || new_frequency > MAX_FREE_FREQ)
+						new_frequency = sanitize_frequency(new_frequency)
+					signaler.set_frequency(new_frequency)
+
+				if(href_list["code"])
+					signaler.code += text2num(href_list["code"])
+					signaler.code = round(signaler.code)
+					signaler.code = min(100, signaler.code)
+					signaler.code = max(1, signaler.code)
+
+			if("directive")
+				if(href_list["getdna"])
+					var/mob/living/M = card.loc
+					var/count = 0
+					while(!isliving(M))
+						if(!M || !M.loc)
+							return 0 //For a runtime where M ends up in nullspace (similar to bluespace but less colourful)
+						M = M.loc
+						count++
+						if(count >= 6)
+							to_chat(src, "You are not being carried by anyone!")
+							return 0
+					spawn CheckDNA(M, src)

```

Comment:
what the shit fuck is this

Reply:
This is the rest of the shitcode. You're gonna find it on the health analyzer software, too (and that _is_ my fault)